aiohttp_jinja2/__init__.py

```diff
@@ -86,7 +86,12 @@ def wrapped(*args):
             if isinstance(context, web.StreamResponse):
                 return context
 
-            request = args[-1]
+            # Supports class based views see web.View
+            if isinstance(args[0], web.View):
```

Comment:
You may want to check against abstract base class: `abc.AbstractView`, user can have own `View` implementation.